front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainScheduleExportCSV.ts

```diff
@@ -0,0 +1,148 @@
+import type { PositionSpeedTime, SpeedPosition, Train } from 'reducers/osrdsimulation/types';
+import * as d3 from 'd3';
+import { BaseOrEcoType } from './DriverTrainScheduleTypes';
+
+/**
+ * CSV Export of trainschedule
+ *
+ * Rows : position in km, speed in km/h, speed limit in km/h, time in s
+ *
+ */
+
+enum CSVKeys {
+  op = 'op',
+  ch = 'ch',
+  lineCode = 'lineCode',
+  trackName = 'trackName',
+  position = 'position',
+  speed = 'speed',
+  speedLimit = 'speedLimit',
+  time = 'time',
+}
+
+type CSVData = {
+  [key in keyof typeof CSVKeys]: string;
+};
+
+type PositionSpeedTimeOP = PositionSpeedTime & {
+  speedLimit?: string | number;
+  op?: '';
+  ch?: '';
+  lineCode?: '';
+  trackName?: '';
+};
+
+const pointToComma = (number: number) => number.toString().replace('.', ',');
+
+const interpolateValue = (
+  position: number,
+  speedList: PositionSpeedTime[],
+  value: 'speed' | 'time'
+): number => {
+  const bisector = d3.bisectLeft(
+    speedList.map((d: SpeedPosition) => d.position),
+    position
+  );
+  if (speedList[bisector - 1]) {
```

Comment:
Early return the easiest case. I think you can also use const for speedList[bisector - 1] and speedList[bisector] to improve the readability
```suggestion
  if (bisector === 0) return speedList[bisector][value];

  const leftSpeed = speedList[bisector - 1];
  const rightSpeed = speedList[bisector];
  
  const totalDistance = rightSpeed.position - leftSpeed.position;
  const distance = position - leftSpeed.position;
  const totalDifference = rightSpeed[value] - leftSpeed[value];
  return leftSpeed[value] + (totalDifference * distance) / totalDistance;
```

Reply:
done, thx!